flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierAligner.java

```diff
@@ -318,6 +326,16 @@ public long getAlignmentDurationNanos() {
 		}
 	}
 
+	public void registerInputGate(CheckpointedInputGate inputGate) {
```

Comment:
Another option is passing `inputGate` in constructor instead to avoid exposing another methods outside. Otherwise in tests if someone constructs the `CheckpointBarrierAligner`, but forgets to register gate afterwards, then the gates are actually touched in the internal related processes to bring unexpected behaviors.